src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -680,15 +624,21 @@ protected void codeToRetry() {
    */
   public static boolean restartMqtt() {
     // Shutdown and cleanup existing MQTT manager
+    boolean restartSuccess = true;
     if (mqttMgr != null) {
-      DefaultEventHandler.delMqttListener(mqttMgr);
-      mqttMgr.disconnect();
-      mqttMgr.stopMqttThread();
-      mqttMgr = null;
+      try {
+        mqttMgr.restart();
+      } catch (Exception e) {
+        restartSuccess = false;
+        Logger.LOG_SERIOUS("An error occurred while restarting the MQTT manager!");
```

Comment:
Why are other uses calling out exception, where as this is calling out error?

Reply:
changed